src/plugins/discover/public/application/embeddable/search_embeddable.ts

```diff
@@ -298,6 +306,20 @@ export class SearchEmbeddable
         this.services.uiSettings.get(SORT_DEFAULT_ORDER_SETTING)
       )
     );
+    if (useNewFieldsApi) {
+      searchSource.removeField('fieldsFromSource');
+      const fields: Record<string, any> = { field: '*' };
```

Comment:
Nit: another `any` that could bite the dust

Reply:
Will address this in the next PR